front/src/modules/ui/board/hooks/useBoardColumns.ts

```diff
@@ -0,0 +1,49 @@
+import { useRecoilState } from 'recoil';
+
+import { boardColumnsState } from '../states/boardColumnsState';
+import { BoardColumnDefinition } from '../types/BoardColumnDefinition';
+
+export const useBoardColumnns = () => {
+  const [boardColumns, setBoardColumns] = useRecoilState(boardColumnsState);
+
+  const handleColumnMove = (
+    direction: string,
+    column: BoardColumnDefinition,
+  ) => {
+    const currentColumnArrayIndex = boardColumns.findIndex(
+      (tableColumn) => tableColumn.id === column.id,
+    );
+
+    const targetColumnArrayIndex =
```

Comment:
this looks like duplicated logic with Table. Can we extract it to a common hook? useMoveViewFields() you can put it into the views folder

Reply:
Done!

Reply:
Are we sure we want to name this `useMoveViewFields`? Board columns are PipelineStages, not ViewFields, unlike Table columns. Couldn't this cause confusion in the future @charlesBochet? We could perhaps name this `useMoveViewColumns` for instance?

Reply:
It's a good point!